sphinx-search-as-you-type.rst

```diff
@@ -0,0 +1,113 @@
+.. post:: July 15, 2019
+   :tags: gsoc, search, feature, in-doc-search
+   :author: Vaibhav
+   :location: LKO
+
+Search Update
+=============
+
+Giving users the ability to easily find the information that they
+are looking for has always been important for Read the Docs.
+This year, I, `Vaibhav Gupta`_, took the opportunity provided
+by Google Summer of Code to improve the search.
+The main goal of my `GSoC project`_ was to enable multiple hits per search result,
+improve the UI/UX around code search and to make an `Sphinx extension`_ to provide
+"search as you type" experience to the users.
+
+Google Summer of Code
+---------------------
+
+Google Summer of Code is a global program where students work with
+an open source organization on a 3-month programming project.
+I got accepted in the program and was excited to spend my summer with Read the Docs.
+
+I have worked full time during the GSoC coding period and implemented various features.
+All of my search related work can be seen in the `In-Doc Search UI Project Board`_.
+
+Background
+----------
+
+Search code was voluntarily contributed by `Rob Hudson`_,
+`back in 2013`_ and then improved by other contributors.
+It was greatly improved and upgraded by `Safwan Rahman`_ during :doc:`GSoC'18 <search-improvements>`.
+Continuining on the same path,
+I have implemented some new features on top of the existing search backend.
+
+New Features
+------------
+
+During the GSoC period, I have worked on the following features:
+
+- `In-Doc Search UI`_: We have worked on a Sphinx extension which adds a clean and minimal
+  search UI to your docs. It supports "search as you type" feature.
+  So, it is now possible to get instant results without being redirected to any other page.
+  Read the docs `here`_.
+
+.. image:: /_static/in-doc-search-demo.gif
+    :width: 100%
+    :target: /_static/in-doc-search-demo.gif
```

Comment:
I love the demo, but I think we should also link to some example searches in our docs for each of these features that show it off.